src/interval.jl

```diff
@@ -187,70 +187,63 @@ Base.isopen(interval::AbstractInterval{T,L,R}) where {T,L,R} = L === Open && R =
 isunbounded(interval::AbstractInterval{T,L,R}) where {T,L,R} = L === Unbounded && R === Unbounded
 isbounded(interval::AbstractInterval{T,L,R}) where {T,L,R} = L !== Unbounded && R !== Unbounded
 
-# Minimum and maximum calls for Closed and Unbounded Intervals, just return first and last.
 function Base.minimum(interval::AbstractInterval{T,L,R}; increment=nothing) where {T,L,R}
-    first(interval) === nothing && return typemin(T)
-    return first(interval)
+    return L === Unbounded ? typemin(T) : first(interval)
 end
 
-function Base.maximum(interval::AbstractInterval{T,L,R}; increment=nothing) where {T,L,R}
-    last(interval) === nothing && return typemax(T)
-    return last(interval)
-end
-
-# Minimum and maximum calls for open bounds and ambigous types, return open side +- increment
-# Default increment val is eps(T)
 function Base.minimum(interval::AbstractInterval{T,Open,R}; increment=eps(T)) where {T,R}
-    isempty(interval) && return nothing
+    isempty(interval) && throw(BoundsError(interval, 0))
     min_val = first(interval)
+    # Since intervals can't have NaN, we can just use !isfinite to check if infinite
     !isfinite(min_val) && return typemin(T)
     min_val + increment ∈ interval && return min_val + increment
```

Comment:
```suggestion
    min_val = interval.first + increment
    # Since intervals can't have NaN, we can just use !isfinite to check if infinite
    !isfinite(min_val) && return typemin(T)
    min_val ∈ interval && return min_val
```